File: function-sdk-python/src/nanofaas/sdk/context.py
import contextvars
import logging

_execution_id = contextvars.ContextVar("execution_id", default=None)
_trace_id = contextvars.ContextVar("trace_id", default=None)

def get_execution_id() -> str | None:
    return _execution_id.get()

def get_trace_id() -> str | None:
    return _trace_id.get()

def set_context(execution_id: str | None, trace_id: str | None):
    _execution_id.set(execution_id)
    _trace_id.set(trace_id)

def get_logger(name: str):
    return logging.getLogger(name)

File: sdks/javascript/src/context.ts
import { AsyncLocalStorage } from "node:async_hooks";

type RequestContext = {
    executionId: string;
    traceId?: string;
};

const requestContext = new AsyncLocalStorage<RequestContext>();

export function runWithContext<T>(context: RequestContext, fn: () => T): T {
    return requestContext.run(context, fn);
}

export function getExecutionId(): string | undefined {
    return requestContext.getStore()?.executionId;
}

export function getTraceId(): string | undefined {
    return requestContext.getStore()?.traceId;
}
